### PR TITLE
Parse log expression

### PR DESCRIPTION
**Pull request by fam-ca**
# Description

# Checklist

- [x] I added any unit tests needed to cover the proposed changes
- [x] All top-level declarations have an explicit type
- [x] All exposed declarations have a Haddock comment
- [x] My changes do not break any existing tests (I ran `stack test` before pushing)

### CODE REVIEW

File: test/Spec.hs [hunk 1]
```diff
@@ -1 +1 @@
-{-# OPTIONS_GHC -F -pgmF hspec-discover #-}
+-- {-# OPTIONS_GHC -F -pgmF hspec-discover #-}
```
Comment (aabounegm) [inline test/Spec.hs]: Why did you comment this out?

File: src/Banana/Syntax/Parser.hs [hunk 2]
```diff
@@ -73,6 +79,20 @@ parseVarAssign = do
...
+-- parseIfStatement = do
+--   string "if" >> space >> whiteSpace
+--   cond <- parseExpr
+--   whiteSpace >> string "then" >> whiteSpace
+--   true <- parseVarAssign
+--   string "else" >> whiteSpace
+--   false <- parseVarAssign
+--   IfStatement (cond <$> (parseExpr <* whiteSpace))
+--     [true <$> (parseVarAssign <* whiteSpace)
+--     , false <$> (parseVarAssign <* whiteSpace)]
```
Comment (aabounegm) [inline src/Banana/Syntax/Parser.hs:93]: Why is this commented out?

File: src/Banana/Syntax/Parser.hs [hunk 3]
```diff
@@ -52,14 +52,20 @@ term = parens parseExpr
...
 table :: [[Operator Parser (Expr a)]]
-table = [ [binary "*" Mul AssocLeft, binary "/" Div AssocLeft ]
-        , [binary "+" Add AssocLeft, binary "-" Sub AssocLeft ]
+table = [ [binary "*"  Mul AssocLeft,  binary "/"  Div AssocLeft ]
+        , [binary "+"  Add AssocLeft,  binary "-"  Sub AssocLeft ]
+        , [binary "and" And AssocLeft, binary "or" Or  AssocLeft]
+        , [prefix "not" Not]
+        , [binary "<"  Less AssocNone, binary ">"  More AssocNone ]
+        , [binary "<=" LEq  AssocNone,  binary ">=" MEq AssocNone ]
+        , [binary "="  Eq   AssocRight,  binary "/=" NotEq AssocRight ]
```
Comment (aabounegm) [inline src/Banana/Syntax/Parser.hs:61]: This messes up the order of operations. Please check the [grammar](https://github.com/aabounegm/banana/wiki/Language-Grammar).

File: test/Banana/Syntax/ParserSpec.hs [hunk 4]
```diff
@@ -41,6 +41,34 @@ spec = do
...
+
+    it "parses \"x>=1\" successfully" $ do
+      "x>=1\n" `shouldParseAsExpr` MEq (Var "x") (Lit 1.0)
+
+    it "parses \"1=1\" successfully" $ do
+      "1=1\n" `shouldParseAsExpr` Eq (Lit 1.0) (Lit 1.0)
+
+    it "parses \"2/=1\" successfully" $ do
+      "2/=1\n" `shouldParseAsExpr` NotEq (Lit 2.0) (Lit 1.0)
+
```
Comment (aabounegm) [inline test/Banana/Syntax/ParserSpec.hs:70]: Some tests that check the oder of operations preservation would be nice

File: src/Banana/Syntax/AST.hs [hunk 5]
```diff
@@ -33,10 +42,17 @@ data VarAssign a = VarAssign
...
+data IfStatement a = If
+  { condition :: [Expr a]
+  , statement :: [Statement a]
+  } deriving (Show, Eq)
+
 -- | A wrapper to collect all statement types together
 data Statement a
   = VarDeclaration (VarDecl a)
   | VarAssignment  (VarAssign a)
+  | IfStatement    (IfStatement a) -- FuncCall       (Function a)
```
Comment (aabounegm) [inline src/Banana/Syntax/AST.hs:55]: What does this comment mean?

**Review by aabounegm — CHANGES_REQUESTED** [review]
What are all of these files under `dist-newstyle`? And why did you create a `tasks.json`?

File: src/Banana/Syntax/Parser.hs (hunk 2 above)
Comment (fam-ca) [reply src/Banana/Syntax/Parser.hs:93]: Because it doesn't work yet

File: src/Banana/Syntax/AST.hs (hunk 5 above)
Comment (fam-ca) [reply src/Banana/Syntax/AST.hs:55]: It's for future changes

File: test/Spec.hs (hunk 1 above)
Comment (fam-ca) [reply test/Spec.hs]: I'll fix it

**Comment by aabounegm** [comment]
Closed in favor of #4

**Pull request closed without merging**